src/main/scala/tech/cryptonomic/conseil/tezos/TezosTypes.scala

```diff
@@ -326,34 +327,37 @@ object TezosTypes {
                                            )
 
   final case class AppliedOperationResultStatus(
-                                   status: String,
-                                   errors: Option[List[String]],
-                                   storage: Option[Any],
-                                   balanceUpdates: Option[AppliedOperationBalanceUpdates],
-                                   originatedContracts: Option[String],
-                                   consumedGas: Option[Int],
-                                   storageSizeDiff: Option[Int]
-                                   )
+    status: String,
+    errors: Option[List[String]],
+    storage: Option[Any],
+    balanceUpdates: Option[AppliedOperationBalanceUpdates],
+    originatedContracts: Option[String],
+    consumedGas: Option[Int],
+    storageSizeDiff: Option[Int]
+  )
 
   final case class AccountDelegate(
-                            setable: Boolean,
-                            value: Option[String]
-                            )
+    setable: Boolean,
+    value: Option[PublicKeyHash]
+  )
+
+  //represents unparsed micheline json
+  final case class AccountScript(code: String) extends AnyVal
 
   final case class Account(
-                    manager: String,
-                    balance: scala.math.BigDecimal,
-                    spendable: Boolean,
-                    delegate: AccountDelegate,
-                    script: Option[String],
-                    counter: Int
-                    )
+    manager: PublicKeyHash,
+    balance: scala.math.BigDecimal,
+    spendable: Boolean,
+    delegate: AccountDelegate,
+    script: Option[AccountScript],
```

Comment:
I really like this pattern of making type synonyms for very specific instances of data. This isn't just an Option of a String, this needs to be an Option of an AccountScript. The benefit of the compiler catching these bugs reduces a lot of overhead, but one thing I wonder is: have we not seen errors of this kind because they've been completely eliminated, or is the compiler not giving us enough information when things like happen indirectly, and if the latter is the case, how do we fix this?

Reply:
Not sure if I get this right, but if you're asking how we forgot to "type" some of the contents with more precise definitions than `String` or `Long`, the answer is that the code is evolving.
Hence we're migrating piece by piece.

I first used this approach to decode block operations, because it was necessary for the decoders to apply specific rules of validation to the content of different fields, based on the wrapping custom type. E.g. when we decode a PublicKeyHash or a BlockHash, the parsing will fail if the field string is not correctly base58check-encoded.

The current PR needs pushed me to switch from jackson to circe, just so that I could apply a custom decoder, to the script field content.
I'll explain because it's important that this is clear.
The script contains a json-object and the parsers won't accept to convert that format to a json-string, because they're syntactically different, right?
So I customized the decoder (with circe is easy to do it if I create a specific target type, like `AccountScript`) that will read a json-object, and then convert it to a string that is put inside the `AccountScript` value class.

Does this answer your doubt?